### DISCUSSION TITLE
Connecting JKBMS, correct baud rate?

### DISCUSSION THREAD

**Discussion by SSonder0815** [Louisvdw/dbus-serialbattery#303] [Q&A] [answered]
Dear all, I installed Louis driver to read out and connect to a JKBMS in my Cerbo GX. USB port is correctly identified. The dameon seems to work properly looking into /data/log/serial-starter/current. But under /data/log/dbus-serialbattery.ttyUSB0/current there is always the message that it cannot find it:
INFO:SerialBattery:Testing Jkbms
ERROR:SerialBattery:>>> ERROR: No reply - returning [len:21/55808]

During my research I stumbled across a case where someone changed from 9600 baud to 115200 which solved the issue. Thus I looked up the baud setting with stty -F /dev/ttyUSB0 and it shows in my case 9600 baud.
When I change the baud setting with stty -F /dev/ttyUSB0 115200, the change is visible for a few seconds and then it is switched back to 9600 by whatever mechanism in the background. I have the suspicion that this could be the reason that Louis driver is not reading out my JKBMS.

Any idea from the community a) if the baud setting could be the issue and if yes b) who knows what is influencing the setting in the background?

**Comment by SSonder0815** [comment]
I searched in the code and found in utils.py the correct 115200 setting information for JKBMS but when I look into the utils.py that has been created with the installation routine on my cerbo GX, the baud settings are not in this file. Why is there a difference between the github code and the one that has been uploaded to the Cerbo GX?

**Comment by Louisvdw** [comment, marked as answer]
Hi

If your JKBMS is not being picked up the most likely problem is issues with your cabling or adapter.
Don't worry about the baud rate of the serial device. The driver will connect to the BMS using the correct settings. After the last release (v0.13) the settings for the batteries have been moved to utils.py (these are used in the betas) and will move again when the gui changes are complete. You should not have to change anything there no matter which version you use.

Check that you do not have any loose connectors or wires between your JKBMS and the USB adapter. You have to connect to the BMS using the port labled GPS. If you are using the JKBMS RS485 adapter that is the connector that the adapter will fit into. Swop the A/B wires from your adapter and USB converter. Are any lights flashing on the USB adapter? 
If you still have issues can you upload pictures of your connections between GX and BMS.

**Comment by SSonder0815** [comment]
Ok for everybody who might run into similar issues.
In my case I figured out that the voltage output of the JK RS485 adapter was below 0.3 V which seemed to be odd. Luckily I had a second one and this one produced a signal output of about 5 V. When I connected this one, the driver recognized the JKBMS in the log file. But there were first error messages in the log file which prohibited that the battery was shown also in the remote console. But after I disabled the driver and enabled it afterwards, the battery was correctly shown in the remote console. Hope this helps other who need to look for some enhanced troubleshooting.
